From 9285caf6b6ae93c2d005ca2ac8cd25b00a6432e7 Mon Sep 17 00:00:00 2001
From: Ranjani Vaidyanathan <Ranjani.Vaidyanathan@freescale.com>
Date: Mon, 9 Jun 2014 11:40:14 -0500
Subject: [PATCH 0412/1074] ENGR00317542-2 ARM:dts:imx6x: Allocate 4K of IRAM
 for DDR freq change code.

Allocate 4K of IRAM space for DDR freq change code in the dts files.
Also reduce the regular IRAM available for other functions by this amount.

Signed-off-by: Ranjani Vaidyanathan <Ranjani.Vaidyanathan@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 arch/arm/boot/dts/imx6dl.dtsi |   10 ++++++++--
 arch/arm/boot/dts/imx6q.dtsi  |   10 ++++++++--
 arch/arm/boot/dts/imx6sx.dtsi |   10 ++++++++--
 3 files changed, 24 insertions(+), 6 deletions(-)

diff --git a/arch/arm/boot/dts/imx6dl.dtsi b/arch/arm/boot/dts/imx6dl.dtsi
index dd35551..f0fa7e8 100644
--- a/arch/arm/boot/dts/imx6dl.dtsi
+++ b/arch/arm/boot/dts/imx6dl.dtsi
@@ -69,9 +69,15 @@
 			clocks = <&clks 142>;
 		};
 
-		ocram: sram@00904000 {
+		ocrams_ddr: sram@00904000 {
+			compatible = "fsl,ddr-lpm-sram";
+			reg = <0x00904000 0x1000>;
+			clocks = <&clks 142>;
+		};
+
+		ocram: sram@00905000 {
 			compatible = "mmio-sram";
-			reg = <0x00904000 0x1C000>;
+			reg = <0x00905000 0x1B000>;
 			clocks = <&clks 142>;
 		};
 
diff --git a/arch/arm/boot/dts/imx6q.dtsi b/arch/arm/boot/dts/imx6q.dtsi
index 06576ac..19a1d5f 100644
--- a/arch/arm/boot/dts/imx6q.dtsi
+++ b/arch/arm/boot/dts/imx6q.dtsi
@@ -102,9 +102,15 @@
 			clocks = <&clks 142>;
 		};
 
-		ocram: sram@00904000 {
+		ocrams_ddr: sram@00904000 {
+			compatible = "fsl,ddr-lpm-sram";
+			reg = <0x00904000 0x1000>;
+			clocks = <&clks 142>;
+		};
+
+		ocram: sram@00905000 {
 			compatible = "mmio-sram";
-			reg = <0x00904000 0x3C000>;
+			reg = <0x00905000 0x3B000>;
 			clocks = <&clks 142>;
 		};
 
diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index f09527b..efa7360 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -117,9 +117,15 @@
 			clocks = <&clks IMX6SX_CLK_OCRAM_S>;
 		};
 
-		ocram: sram@00900000 {
+		ocrams_ddr: sram@00900000 {
+			compatible = "fsl,ddr-lpm-sram";
+			reg = <0x00900000 0x1000>;
+			clocks = <&clks IMX6SX_CLK_OCRAM>;
+		};
+
+		ocram: sram@00901000 {
 			compatible = "mmio-sram";
-			reg = <0x00900000 0x20000>;
+			reg = <0x00901000 0x1F000>;
 			clocks = <&clks IMX6SX_CLK_OCRAM>;
 		};
 
-- 
1.7.5.4

